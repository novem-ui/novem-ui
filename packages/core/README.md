# @novem-ui/core

This lib exports all the Novem UI components