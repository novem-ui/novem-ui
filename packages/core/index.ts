export { default as NovemProvider } from './src/novem-provider'

export { default as Badge } from '@novem-ui/badge'
export * from '@novem-ui/badge'

export { default as Button, ButtonProps } from '@novem-ui/button'

export { default as Switch, SwitchProps } from '@novem-ui/switch'

export { Heading, Label, Paragraph, HeadingProps, LabelProps, ParagraphProps } from '@novem-ui/text'

export { Input, TextField, TextFieldProps } from '@novem-ui/input'

export { default as theme } from '@novem-ui/theme'
export type { Theme } from '@novem-ui/theme'
