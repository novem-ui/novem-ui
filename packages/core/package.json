{
  "name": "@novem-ui/core",
  "version": "0.0.1-alpha.1",
  "description": "Core components",
  "homepage": "https://github.com/jdrets/novem-ui/tree/main/packages/core#readme",
  "license": "ISC",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/jdrets/novem-ui.git"
  },
  "publishConfig": {
    "access": "public",
    "directory": "../../dist/core"
  },
  "scripts": {
    "test": "jest --env=jsdom --passWithNoTests",
    "build": "yarn build:esm && yarn build:types && yarn build:package-json",
    "build:esm": "../../node_modules/.bin/babel . --root-mode upward --extensions .ts,.tsx -d ../../dist/core --source-maps",
    "build:types": "tsc -p ./tsconfig.types.json",
    "build:package-json": "../../scripts/create-package.js"
  },
  "bugs": {
    "url": "https://github.com/jdrets/novem-ui/issues"
  },
  "dependencies": {
    "@novem-ui/badge": "^0.0.1-alpha.1",
    "@novem-ui/theme": "^0.0.1-alpha.1"
  },
  "devDependencies": {
    "@types/novem-theme": "^0.0.1-alpha.1",
    "@types/react": "^17.0.3"
  },
  "peerDependencies": {
    "@emotion/react": "^11.1.5",
    "@emotion/styled": "^11.3.0",
    "polished": "^4.1.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
}
