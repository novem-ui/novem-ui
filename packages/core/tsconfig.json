{
  "extends": "../../tsconfig.json",
  "files": [],
  "include": ["**/*.ts", "**/*.tsx", "../../emotion.d.ts"],
  "exclude": ["node_modules/**", "dist/**"]
}
