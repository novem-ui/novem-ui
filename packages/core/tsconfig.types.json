{
  "extends": "../../tsconfig.json",
  "files": [],
  "compilerOptions": {
    "emitDeclarationOnly": true,
    "declaration": true,
    "declarationDir": "../../dist/",
    "declarationMap": true,
    "types": ["novem-theme"]
  },
  "include": ["**/*.ts", "**/*.tsx"],
  "exclude": ["node_modules/**", "dist/**", "**/*.spec.tsx", "**/*.stories.tsx"]
}
