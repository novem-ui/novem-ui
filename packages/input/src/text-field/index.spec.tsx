import React, { PropsWithRef } from 'react'

import { act, render, screen, waitFor } from '@testing-library/react'
import { matchers } from '@emotion/jest'
import { NovemProvider } from '@novem-ui/core'
import userEvent from '@testing-library/user-event'

import TextField, { TextFieldProps } from '.'

beforeAll(() => {
  expect.extend(matchers)
})

const setup = (props: PropsWithRef<TextFieldProps> = {}) => {
  const innerProps = { onChange: () => null, ...props }
  const result = render(
    <NovemProvider>
      <TextField {...innerProps} />
    </NovemProvider>
  )

  return result
}

describe('input/text-field', () => {
  test('should render a TextField', () => {
    const { container } = setup({ value: 'Default value' })

    expect(container).toMatchSnapshot()
  })

  test('should render a TextField with label', () => {
    const { container } = setup({ value: 'Default value', label: 'TextField Label' })

    expect(container).toMatchSnapshot()
  })

  test('should render a TextField with error', () => {
    const { container } = setup({ value: 'Default value', error: new Error('error message') })

    expect(container).toMatchSnapshot()
  })

  test('should render a TextField with a char counter', () => {
    const { container } = setup({ value: 'Default value', count: true })

    expect(container).toMatchSnapshot()
    expect(screen.queryByText('13')).toBeInTheDocument()
  })

  test('should update the char counter on change', async () => {
    const onChange = jest.fn()
    const { container } = setup({
      label: 'Input element',
      value: 'value',
      count: true,
      onChange
    })

    expect(screen.queryByText('5')).toBeInTheDocument()

    act(() => userEvent.type(container.querySelector('input'), 'valuee'))

    await waitFor(() => expect(screen.queryByText('6')).toBeInTheDocument())

    await waitFor(() => expect(onChange).toBeCalled())
  })

  test('should render a TextField with a limit', () => {
    const { container } = setup({ value: 'value', maxLength: 10 })

    expect(container).toMatchSnapshot()
    expect(screen.queryByText('5/10')).toBeInTheDocument()
  })

  test('should render a TextField with a helper', () => {
    const { container } = setup({ value: 'value', helper: 'This is a helper text' })

    expect(container).toMatchSnapshot()
    expect(screen.queryByText('This is a helper text')).toBeInTheDocument()
  })

  test('should render a TextField with an error message (from string)', () => {
    const { container } = setup({ value: 'value', error: 'This is an error message' })

    expect(container).toMatchSnapshot()
    expect(screen.queryByText('This is an error message')).toBeInTheDocument()
  })

  test('should render a TextField with an error message (from error object)', () => {
    const { container } = setup({ value: 'value', error: new Error('This is an error message') })

    expect(container).toMatchSnapshot()
    expect(screen.queryByText('This is an error message')).toBeInTheDocument()
  })

  test('should render a TextField with a valid state', () => {
    const { container } = setup({ value: 'value', valid: true })

    expect(container).toMatchSnapshot()
  })

  test('should render a disabled TextField', () => {
    const { container } = setup({ value: 'value', disabled: true })

    expect(container).toMatchSnapshot()
  })
})
