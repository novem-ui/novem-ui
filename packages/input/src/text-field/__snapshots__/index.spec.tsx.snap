// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input/text-field should render a TextField 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: 1px solid rgba(0, 0, 0, .25);
  border-radius: 1rem;
  box-sizing: border-box;
  height: 3rem;
  font-size: 16px;
  padding: 1rem;
  -webkit-transition: border ease-in 0.25s;
  transition: border ease-in 0.25s;
}

.emotion-2:focus,
.emotion-2:hover {
  outline: unset;
  border-color: #F1694B;
  box-shadow: 0 2px 4px 0 rgba(241,105,75,0.15);
}

.emotion-2:disabled {
  pointer-events: none;
  background-color: rgba(154,152,151,0.15);
}

.emotion-2:-moz-read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

.emotion-2:read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

<div>
  <div
    class="emotion-0 emotion-1"
    value="Default value"
  >
    <div
      style="position: relative; display: flex; width: 100%;"
    >
      <input
        class="emotion-2 emotion-3"
        value="Default value"
      />
    </div>
  </div>
</div>
`;

exports[`input/text-field should render a TextField with a char counter 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: 1px solid rgba(0, 0, 0, .25);
  border-radius: 1rem;
  box-sizing: border-box;
  height: 3rem;
  font-size: 16px;
  padding: 1rem;
  -webkit-transition: border ease-in 0.25s;
  transition: border ease-in 0.25s;
}

.emotion-2:focus,
.emotion-2:hover {
  outline: unset;
  border-color: #F1694B;
  box-shadow: 0 2px 4px 0 rgba(241,105,75,0.15);
}

.emotion-2:disabled {
  pointer-events: none;
  background-color: rgba(154,152,151,0.15);
}

.emotion-2:-moz-read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

.emotion-2:read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

.emotion-4 {
  color: rgba(154,152,151,0.85);
  font-size: 12px;
  margin: 0;
  text-align: right;
  margin-left: 0.5rem;
  margin-right: 0.5rem;
  margin-top: 0.25rem;
}

<div>
  <div
    class="emotion-0 emotion-1"
    value="Default value"
  >
    <div
      style="position: relative; display: flex; width: 100%;"
    >
      <input
        class="emotion-2 emotion-3"
        value="Default value"
      />
    </div>
    <div
      style="display: flex;"
    >
      <div
        style="flex: 1;"
      />
      <p
        class="emotion-4 emotion-5"
        color="grey.400"
      >
        13
      </p>
    </div>
  </div>
</div>
`;

exports[`input/text-field should render a TextField with a helper 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: 1px solid rgba(0, 0, 0, .25);
  border-radius: 1rem;
  box-sizing: border-box;
  height: 3rem;
  font-size: 16px;
  padding: 1rem;
  -webkit-transition: border ease-in 0.25s;
  transition: border ease-in 0.25s;
}

.emotion-2:focus,
.emotion-2:hover {
  outline: unset;
  border-color: #F1694B;
  box-shadow: 0 2px 4px 0 rgba(241,105,75,0.15);
}

.emotion-2:disabled {
  pointer-events: none;
  background-color: rgba(154,152,151,0.15);
}

.emotion-2:-moz-read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

.emotion-2:read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

.emotion-4 {
  color: rgba(154,152,151,0.85);
  font-size: 12px;
  margin: 0;
  margin-left: 0.5rem;
  margin-top: 0.25rem;
}

<div>
  <div
    class="emotion-0 emotion-1"
    value="value"
  >
    <div
      style="position: relative; display: flex; width: 100%;"
    >
      <input
        class="emotion-2 emotion-3"
        value="value"
      />
    </div>
    <div
      style="display: flex;"
    >
      <div
        style="flex: 1;"
      >
        <p
          class="emotion-4 emotion-5"
          color="grey.400"
        >
          This is a helper text
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`input/text-field should render a TextField with a limit 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: 1px solid rgba(0, 0, 0, .25);
  border-radius: 1rem;
  box-sizing: border-box;
  height: 3rem;
  font-size: 16px;
  padding: 1rem;
  -webkit-transition: border ease-in 0.25s;
  transition: border ease-in 0.25s;
}

.emotion-2:focus,
.emotion-2:hover {
  outline: unset;
  border-color: #F1694B;
  box-shadow: 0 2px 4px 0 rgba(241,105,75,0.15);
}

.emotion-2:disabled {
  pointer-events: none;
  background-color: rgba(154,152,151,0.15);
}

.emotion-2:-moz-read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

.emotion-2:read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

.emotion-4 {
  color: rgba(154,152,151,0.85);
  font-size: 12px;
  margin: 0;
  text-align: right;
  margin-left: 0.5rem;
  margin-right: 0.5rem;
  margin-top: 0.25rem;
}

<div>
  <div
    class="emotion-0 emotion-1"
    value="value"
  >
    <div
      style="position: relative; display: flex; width: 100%;"
    >
      <input
        class="emotion-2 emotion-3"
        maxlength="10"
        value="value"
      />
    </div>
    <div
      style="display: flex;"
    >
      <div
        style="flex: 1;"
      />
      <p
        class="emotion-4 emotion-5"
        color="grey.400"
      >
        5
        /10
      </p>
    </div>
  </div>
</div>
`;

exports[`input/text-field should render a TextField with a valid state 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: 1px solid rgba(0, 0, 0, .25);
  border-radius: 1rem;
  border-color: #59BC6F;
  box-sizing: border-box;
  height: 3rem;
  font-size: 16px;
  padding: 1rem;
  -webkit-transition: border ease-in 0.25s;
  transition: border ease-in 0.25s;
}

.emotion-2:focus,
.emotion-2:hover {
  outline: unset;
  border-color: #F1694B;
  border-color: #59BC6F;
  box-shadow: 0 2px 4px 0 rgba(241,105,75,0.15);
  box-shadow: 0 2px 4px 0 rgba(89,188,111,0.15);
}

.emotion-2:disabled {
  pointer-events: none;
  background-color: rgba(154,152,151,0.15);
}

.emotion-2:-moz-read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

.emotion-2:read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

.emotion-4 {
  top: 1.5rem;
  right: 1rem;
  position: absolute;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #59BC6F;
  border-radius: 0.625rem;
  height: 1.25rem;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 1.25rem;
}

.emotion-4 .i-icon {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
  <div
    class="emotion-0 emotion-1"
    value="value"
  >
    <div
      style="position: relative; display: flex; width: 100%;"
    >
      <input
        class="emotion-2 emotion-3"
        value="value"
      />
      <div
        class="emotion-4 emotion-5"
      >
        <span
          class="i-icon i-icon-check"
        >
          <svg
            fill="none"
            height="10"
            viewBox="0 0 48 48"
            width="10"
          >
            <path
              d="M43 11L16.875 37L5 25.1818"
              stroke="#fff"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="8"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`input/text-field should render a TextField with an error message (from error object) 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: 1px solid rgba(0, 0, 0, .25);
  border-radius: 1rem;
  border-color: #EB5757;
  box-sizing: border-box;
  height: 3rem;
  font-size: 16px;
  padding: 1rem;
  -webkit-transition: border ease-in 0.25s;
  transition: border ease-in 0.25s;
}

.emotion-2:focus,
.emotion-2:hover {
  outline: unset;
  border-color: #F1694B;
  border-color: #EB5757;
  box-shadow: 0 2px 4px 0 rgba(241,105,75,0.15);
  box-shadow: 0 2px 4px 0 rgba(235,87,87,0.15);
}

.emotion-2:disabled {
  pointer-events: none;
  background-color: rgba(154,152,151,0.15);
}

.emotion-2:-moz-read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

.emotion-2:read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

.emotion-4 {
  top: 1.5rem;
  right: 1rem;
  position: absolute;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #EB5757;
  border-radius: 0.625rem;
  height: 1.25rem;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 1.25rem;
}

.emotion-4 .i-icon {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-6 {
  color: #EB5757;
  font-size: 12px;
  margin: 0;
  margin-left: 0.5rem;
  margin-top: 0.25rem;
}

<div>
  <div
    class="emotion-0 emotion-1"
    value="value"
  >
    <div
      style="position: relative; display: flex; width: 100%;"
    >
      <input
        class="emotion-2 emotion-3"
        value="value"
      />
      <div
        class="emotion-4 emotion-5"
      >
        <span
          class="i-icon i-icon-close"
        >
          <svg
            fill="none"
            height="10"
            viewBox="0 0 48 48"
            width="10"
          >
            <path
              d="M8 8L40 40"
              stroke="#fff"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="8"
            />
            <path
              d="M8 40L40 8"
              stroke="#fff"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="8"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      style="display: flex;"
    >
      <div
        style="flex: 1;"
      >
        <p
          class="emotion-6 emotion-7"
          color="red.500"
        >
          This is an error message
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`input/text-field should render a TextField with an error message (from string) 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: 1px solid rgba(0, 0, 0, .25);
  border-radius: 1rem;
  border-color: #EB5757;
  box-sizing: border-box;
  height: 3rem;
  font-size: 16px;
  padding: 1rem;
  -webkit-transition: border ease-in 0.25s;
  transition: border ease-in 0.25s;
}

.emotion-2:focus,
.emotion-2:hover {
  outline: unset;
  border-color: #F1694B;
  border-color: #EB5757;
  box-shadow: 0 2px 4px 0 rgba(241,105,75,0.15);
  box-shadow: 0 2px 4px 0 rgba(235,87,87,0.15);
}

.emotion-2:disabled {
  pointer-events: none;
  background-color: rgba(154,152,151,0.15);
}

.emotion-2:-moz-read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

.emotion-2:read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

.emotion-4 {
  top: 1.5rem;
  right: 1rem;
  position: absolute;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #EB5757;
  border-radius: 0.625rem;
  height: 1.25rem;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 1.25rem;
}

.emotion-4 .i-icon {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-6 {
  color: #EB5757;
  font-size: 12px;
  margin: 0;
  margin-left: 0.5rem;
  margin-top: 0.25rem;
}

<div>
  <div
    class="emotion-0 emotion-1"
    value="value"
  >
    <div
      style="position: relative; display: flex; width: 100%;"
    >
      <input
        class="emotion-2 emotion-3"
        value="value"
      />
      <div
        class="emotion-4 emotion-5"
      >
        <span
          class="i-icon i-icon-close"
        >
          <svg
            fill="none"
            height="10"
            viewBox="0 0 48 48"
            width="10"
          >
            <path
              d="M8 8L40 40"
              stroke="#fff"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="8"
            />
            <path
              d="M8 40L40 8"
              stroke="#fff"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="8"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      style="display: flex;"
    >
      <div
        style="flex: 1;"
      >
        <p
          class="emotion-6 emotion-7"
          color="red.500"
        >
          This is an error message
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`input/text-field should render a TextField with error 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: 1px solid rgba(0, 0, 0, .25);
  border-radius: 1rem;
  border-color: #EB5757;
  box-sizing: border-box;
  height: 3rem;
  font-size: 16px;
  padding: 1rem;
  -webkit-transition: border ease-in 0.25s;
  transition: border ease-in 0.25s;
}

.emotion-2:focus,
.emotion-2:hover {
  outline: unset;
  border-color: #F1694B;
  border-color: #EB5757;
  box-shadow: 0 2px 4px 0 rgba(241,105,75,0.15);
  box-shadow: 0 2px 4px 0 rgba(235,87,87,0.15);
}

.emotion-2:disabled {
  pointer-events: none;
  background-color: rgba(154,152,151,0.15);
}

.emotion-2:-moz-read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

.emotion-2:read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

.emotion-4 {
  top: 1.5rem;
  right: 1rem;
  position: absolute;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #EB5757;
  border-radius: 0.625rem;
  height: 1.25rem;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 1.25rem;
}

.emotion-4 .i-icon {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-6 {
  color: #EB5757;
  font-size: 12px;
  margin: 0;
  margin-left: 0.5rem;
  margin-top: 0.25rem;
}

<div>
  <div
    class="emotion-0 emotion-1"
    value="Default value"
  >
    <div
      style="position: relative; display: flex; width: 100%;"
    >
      <input
        class="emotion-2 emotion-3"
        value="Default value"
      />
      <div
        class="emotion-4 emotion-5"
      >
        <span
          class="i-icon i-icon-close"
        >
          <svg
            fill="none"
            height="10"
            viewBox="0 0 48 48"
            width="10"
          >
            <path
              d="M8 8L40 40"
              stroke="#fff"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="8"
            />
            <path
              d="M8 40L40 8"
              stroke="#fff"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="8"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      style="display: flex;"
    >
      <div
        style="flex: 1;"
      >
        <p
          class="emotion-6 emotion-7"
          color="red.500"
        >
          error message
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`input/text-field should render a TextField with label 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  font-size: 12px;
  margin: 0;
  margin-left: 0.5rem;
  margin-bottom: 0.25rem;
}

.emotion-4 {
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: 1px solid rgba(0, 0, 0, .25);
  border-radius: 1rem;
  box-sizing: border-box;
  height: 3rem;
  font-size: 16px;
  padding: 1rem;
  -webkit-transition: border ease-in 0.25s;
  transition: border ease-in 0.25s;
}

.emotion-4:focus,
.emotion-4:hover {
  outline: unset;
  border-color: #F1694B;
  box-shadow: 0 2px 4px 0 rgba(241,105,75,0.15);
}

.emotion-4:disabled {
  pointer-events: none;
  background-color: rgba(154,152,151,0.15);
}

.emotion-4:-moz-read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

.emotion-4:read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

<div>
  <div
    class="emotion-0 emotion-1"
    value="Default value"
  >
    <label
      class="emotion-2 emotion-3"
      style="display: block;"
    >
      TextField Label
    </label>
    <div
      style="position: relative; display: flex; width: 100%;"
    >
      <input
        class="emotion-4 emotion-5"
        value="Default value"
      />
    </div>
  </div>
</div>
`;

exports[`input/text-field should render a disabled TextField 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: 1px solid rgba(0, 0, 0, .25);
  border-radius: 1rem;
  box-sizing: border-box;
  height: 3rem;
  font-size: 16px;
  padding: 1rem;
  -webkit-transition: border ease-in 0.25s;
  transition: border ease-in 0.25s;
}

.emotion-2:focus,
.emotion-2:hover {
  outline: unset;
  border-color: #F1694B;
  box-shadow: 0 2px 4px 0 rgba(241,105,75,0.15);
}

.emotion-2:disabled {
  pointer-events: none;
  background-color: rgba(154,152,151,0.15);
}

.emotion-2:-moz-read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

.emotion-2:read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

<div>
  <div
    class="emotion-0 emotion-1"
    value="value"
  >
    <div
      style="position: relative; display: flex; width: 100%;"
    >
      <input
        class="emotion-2 emotion-3"
        disabled=""
        value="value"
      />
    </div>
  </div>
</div>
`;
